api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TPTEligiblePatientListCohortQueries.java

```diff
@@ -1028,209 +1131,929 @@ public CohortDefinition getIPTB1(
+            + "                         ) "
+            + "GROUP BY p.patient_id";
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+
+    sqlCohortDefinition.setQuery(sb.replace(query));
+
+    return sqlCohortDefinition;
+  }
+
+  /**
+   * <b>IMER1</b>:User Story TPT Eligible Patient List <br>
+   *
+   * <ul>
+   *   <li>B1.2: Select all patients with Ficha Resumo (encounter type 53) with Última
+   *       profilaxia(concept id 23985) value coded INH(concept id 656) and Data Fim da Profilaxia
+   *       TPT(value datetime, concept id 6129) and value datetime not null and and between (date
+   *       from Y+ 173 DAYs) and (date from Y + 365 DAYs)
+   * </ul>
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getIPTB1part2() {
+
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    sqlCohortDefinition.setName(" all patients with Ultima profilaxia Isoniazida (Data Fim) B1.2");
+    sqlCohortDefinition.addParameter(new Parameter("endDate", "Before Date", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("startDate", "After Date", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("location", "Location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("53", hivMetadata.getMasterCardEncounterType().getEncounterTypeId());
+    map.put("23985", tbMetadata.getRegimeTPTConcept().getConceptId());
+    map.put("656", tbMetadata.getIsoniazidConcept().getConceptId());
+    map.put("6129", hivMetadata.getDataFinalizacaoProfilaxiaIsoniazidaConcept().getConceptId());
+    map.put("6128", hivMetadata.getDataInicioProfilaxiaIsoniazidaConcept().getConceptId());
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("6122", tbMetadata.getIsoniazidConcept().getConceptId());
+    map.put("1256", hivMetadata.getStartDrugs().getConceptId());
+    map.put("165308", tbMetadata.getDataEstadoDaProfilaxiaConcept().getConceptId());
+    map.put("9", hivMetadata.getPediatriaSeguimentoEncounterType().getEncounterTypeId());
+    map.put("60", tbMetadata.getRegimeTPTEncounterType().getEncounterTypeId());
+    map.put("23982", tbMetadata.getIsoniazidePiridoxinaConcept().getConceptId());
+    map.put("23987", hivMetadata.getPatientTreatmentFollowUp().getConceptId());
+    map.put("1705", hivMetadata.getRestartConcept().getConceptId());
+    map.put("1257", hivMetadata.getContinueRegimenConcept().getConceptId());
+
+    String query =
+        "SELECT pp.patient_id FROM patient pp INNER JOIN encounter e ON pp.patient_id = e.patient_id "
+            + "                                     INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "                                     INNER JOIN obs o2 ON e.encounter_id = o2.encounter_id "
+            + "WHERE pp.voided =0 AND e.voided = 0 AND o.voided=0 AND o2.voided=0 "
```

Comment:
Same here, let's improve this query

Reply:
This query was reimplemented on TPT Eligible ticket.